scipy/stats/_hypotests.py

```diff
@@ -130,3 +131,186 @@ def epps_singleton_2samp(x, y, t=(0.4, 0.8)):
     p = chi2.sf(w, r)
 
     return Epps_Singleton_2sampResult(w, p)
+
+
+PoissonMeansTestResult = namedtuple('PoissonMeansTestResult', ('statistic', 'pvalue'))
+
+
+def poisson_means_test(count1, nobs1, count2, nobs2, diff=0, alternative='two-sided'):
+    r"""
+    Calculate E-test for the mean difference of two samples that follow a Poisson
+    distribution from descriptive statistics
+
+    Let :math:`X_{11},...,X_{1n_1}` and :math:`X_{21},...,X_{2n_2}` be independent
+    samples respectively, from :math:`Poisson(\lambda_1)` and :math:`Poisson(\lambda_2)` distributions. It is well known that
+    
+    .. math:: X_1 = \sum_{i=1}^{n_1} X_{1i} \sim Poisson(n_1\lambda_1)
+    
+    independently of
+    
+    .. math:: X_2 = \sum_{i=1}^{n_2} X_{2i} \sim Poisson(n_2\lambda_2)
+    
+    Let `count1` and `count1` be the observed values of :math:`X_1` and :math:`X_2`,
+    respectively. The problem of interest here is to test
+    
+    .. math::
+       H_0: \lambda_1 - \lambda_2 \le \mathtt{diff} \quad vs. \quad
```

Comment:
Maybe state two-sided test as it is the default, i.e. = diff vs \ne diff. 

If you say after that that alternative hypothesis can be changed to greater and less, it should be clear how the test works in that case